sdk/core/core-client-rest/src/restError.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PathUncheckedResponse } from "./getClient";
+import { RestError, PipelineResponse, createHttpHeaders } from "@azure/core-rest-pipeline";
+
+/**
+ * Creates a rest error from a PathUnchecked response
+ */
+export function createRestError(message: string, response: PathUncheckedResponse): RestError {
+  return new RestError(message, {
+    statusCode: statusCodeToNumber(response.status),
+    request: response.request,
+    response: toPipelineresponse(response),
+  });
+}
+
+function toPipelineresponse(response: PathUncheckedResponse): PipelineResponse {
+  return {
+    headers: createHttpHeaders(response.headers),
+    request: response.request,
+    status: statusCodeToNumber(response.status) ?? -1,
+  };
+}
+
+function statusCodeToNumber(statusCode: string) {
```

Comment:
is the linter running?

```suggestion
function statusCodeToNumber(statusCode: string): number | undefined {
```